### PR TITLE
fix(desktop): workspace chat opens with custom preset

### PR DESCRIPTION
**Pull request by vivishko**
## Description

Fixes a regression where new workspace launches could still start Codex with default high reasoning flags after users changed presets to medium.

Also resolves stale command-state stickiness by auto-reconciling mismatched command/promptCommand pairs (e.g., command=medium with old promptCommand=high --) during both preset save and runtime resolution.

## Key changes:

- Added sync from terminal presets to agent presets for launch-time resolution:
  - `agent-backed-agent-presets.ts`
- Wired this into settings resolution so getAgentPresets and workspace launch paths receive hydrated values:
  - `index.ts`
- Added defensive handling for stale promptCommand values (old high -- paired with new command):
  - `agent-settings.ts`
- Ensured promptCommand is auto-synced when only command is updated in agent settings:
  - `agent-preset-router.utils.ts`
- Invalidated all relevant settings queries after agent preset updates to avoid stale UI/runtime cache:
  - `AgentCard.tsx`
- Refactored shared command derivation logic into a single utility:
  - `terminal-agent-command.ts`
- Removed duplicate preset-name-to-agent-id mapping helper:
  - `agent-preset-name.ts`

## Type of Change

- [x] Bug fix
- [ ] New feature
- [ ] Documentation
- [ ] Refactor
- [ ] Other (please describe):

## Testing

Done:
- Added automated tests covering preset-to-agent command hydration and prompt-command sync edge cases.
- Added regression tests for Codex preset override flow (high -> medium) during workspace launch.
- Included automated unit tests for settings hydration and command normalization paths.

Manual verification:
- Update Codex preset from `high` to `medium`
- Create a new workspace
- Verify launched Codex command uses `model_reasoning_effort="medium"` instead of `"high"`

## Screenshots 

Presets settings:
image

Created codex chat:
image

## Additional Notes

- `--` suffix in Codex prompt command is expected CLI prompt-passing syntax; it is not the reasoning level.
- Fix handles both:
  - newly edited presets,
  - previously saved stale combinations (`command=medium`, `promptCommand=high --`).
- Behavior preserves explicit user `promptCommand` overrides (auto-sync applies only when `promptCommand` is not explicitly provided).

---
## Summary by cubic
Fixes workspace chats launching with the wrong reasoning level after switching presets. New chats and auto-apply launches now use the selected custom preset, and agent/terminal presets stay in sync.

- **Bug Fixes**
  - Hydrates terminal agent configs from terminal presets at launch and in `getAgentPresets`, so `model_reasoning_effort="medium"` is applied.
  - Keeps built‑in terminal presets aligned with agent overrides until a user edits the preset.
  - Preserves explicit `promptCommand` overrides; derives it from `command` when missing and repairs stale defaults when `command` changes.
  - Hydrates `getTerminalPresets` and trigger‑based selections (`getWorkspaceCreationPresets`, `getNewTabPresets`) so auto‑apply launches use overridden commands; invalidates these queries on update/reset to avoid stale caches.

- **Refactors**
  - Centralized command‑to‑prompt derivation in `shared/utils/terminal-agent-command`.
  - Centralized preset‑name→agent‑id mapping in `agent-preset-name`.

<sup>Written for commit 4903c583c1ae580bf1d01450dff5c7e55662f754. Summary will update on new commits.</sup>

## Summary by CodeRabbit

* **New Features**
  * Terminal and agent presets now auto-sync so terminal commands and prompt commands reflect agent definitions while preserving user-overridden fields.
  * Preset retrieval now consistently returns hydrated presets and exposes sensible defaults.
  * Prompt commands are auto-derived when command changes and no prompt is provided.

* **Bug Fixes**
  * Improved consistency and cache refresh behavior across settings and UI flows (invalidate related preset caches on updates/resets).

* **Tests**
  * Added tests covering sync behavior, override preservation, and prompt-command derivation.

### CODE REVIEW

File: apps/desktop/src/shared/utils/agent-settings.test.ts [hunk 1]
```diff
@@ -60,6 +60,56 @@ describe("resolveAgentConfigs", () => {
...
+				],
+			},
+		}).find((preset) => preset.id === "codex");
+
+		expect(codex).toMatchObject({
+			id: "codex",
+			kind: "terminal",
+			command:
+				'codex -c model_reasoning_effort="medium" --dangerously-bypass-approvals-and-sandbox -c model_reasoning_summary="detailed" -c model_supports_reasoning_summaries=true',
+			promptCommand:
```
[the remark this answers is not in the document]
Comment (vivishko) [reply apps/desktop/src/shared/utils/agent-settings.test.ts:83]: fixed with https://github.com/superset-sh/superset/pull/2693/commits/f0e1c013ba1891e2cc8d248a4f2c20949712e78f

File: apps/desktop/src/lib/trpc/routers/settings/index.ts [hunk 2]
```diff
@@ -136,11 +138,24 @@ function saveAgentPresetOverrides(overrides: AgentPresetOverrideEnvelope) {
...
+}
+
+function getHydratedTerminalPresets(
+	presets: TerminalPreset[],
+): TerminalPreset[] {
+	return hydrateAgentBackedPresetCommands({
+		presets,
+		agentPresets: getHydratedAgentPresets(),
+	});
 }
```
[the remark this answers is not in the document]
Comment (vivishko) [reply apps/desktop/src/lib/trpc/routers/settings/index.ts:159]: fixed in https://github.com/superset-sh/superset/pull/2693/commits/60fd07f9f4f7a4956bd6847b5703131893f27911
